sumneko-lua-language-server build fails on a Darwin system

### Describe the bug

Couldn't build sumneko-lua-language-server successfully.

<details>
<summary>Log</summary>
<div>

```
this derivation will be built:
  /nix/store/ssxb50ahrdw2034j4vbbf1wpmka9mg5p-sumneko-lua-language-server-2.5.6.drv
building '/nix/store/ssxb50ahrdw2034j4vbbf1wpmka9mg5p-sumneko-lua-language-server-2.5.6.drv'...
unpacking sources
unpacking source archive /nix/store/8v5mdkz708wcarf093rwnla7cv24va2j-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: -j8 -l8 -fcompile/ninja/macos.ninja
[11/58] Compile C++ build/macos/obj/source_bee/fsevent_osx.objKjK
FAILED: build/macos/obj/source_bee/fsevent_osx.obj
clang++ -MMD -MT build/macos/obj/source_bee/fsevent_osx.obj -MF build/macos/obj/source_bee/fsevent_osx.obj.d -std=c++17 -O2 -Wall -Werror -fvisibility=hidden -mmacosx-version-min=10.12 -I3rd/bee.lua/bee/nonstd -I3rd/bee.lua -DBEE_INLINE -DNDEBUG -Wunguarded-availability -o build/macos/obj/source_bee/fsevent_osx.obj -c 3rd/bee.lua/bee/fsevent/fsevent_osx.cpp
In file included from 3rd/bee.lua/bee/fsevent/fsevent_osx.cpp:2:
In file included from 3rd/bee.lua/bee/format.h:6:
In file included from 3rd/bee.lua/bee/nonstd/fmt/format.h:44:
3rd/bee.lua/bee/nonstd/fmt/core.h:1433:48: error: expected unqualified-id
template <typename T> int check(unformattable) {
                                               ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1558:15: error: no matching function for call to 'make_arg'
        data_{detail::make_arg<
              ^~~~~~~~~~~~~~~~~
3rd/bee.lua/bee/nonstd/fmt/core.h:1597:10: note: in instantiation of member function 'fmt::v7::format_arg_store<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<char>, char>, int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>::format_arg_store' requested here
  return {args...};
         ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1875:28: note: in instantiation of function template specialization 'fmt::v7::make_args_checked<int &, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> &, char [14], char>' requested here
  const auto& vargs = fmt::make_args_checked<Args...>(format_str, args...);
                           ^
3rd/bee.lua/bee/fsevent/fsevent_osx.cpp:156:26: note: in instantiation of function template specialization 'fmt::v7::format<char [14], int &, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> &, char, 0>' requested here
                    std::format("add `{}` `{}`", arg.m_id, arg.m_path)
                         ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1426:41: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'Context'
FMT_CONSTEXPR basic_format_arg<Context> make_arg(const T& value) {
                                        ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1558:15: error: no matching function for call to 'make_arg'
        data_{detail::make_arg<
              ^~~~~~~~~~~~~~~~~
3rd/bee.lua/bee/nonstd/fmt/core.h:1426:41: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'Context'
FMT_CONSTEXPR basic_format_arg<Context> make_arg(const T& value) {
                                        ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1558:15: error: no matching function for call to 'make_arg'
        data_{detail::make_arg<
              ^~~~~~~~~~~~~~~~~
3rd/bee.lua/bee/nonstd/fmt/core.h:1597:10: note: in instantiation of member function 'fmt::v7::format_arg_store<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<char>, char>, int>::format_arg_store' requested here
  return {args...};
         ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1875:28: note: in instantiation of function template specialization 'fmt::v7::make_args_checked<int &, char [12], char>' requested here
  const auto& vargs = fmt::make_args_checked<Args...>(format_str, args...);
                           ^
3rd/bee.lua/bee/fsevent/fsevent_osx.cpp:163:26: note: in instantiation of function template specialization 'fmt::v7::format<char [12], int &, char, 0>' requested here
                    std::format("remove `{}`", arg.m_id)
                         ^
3rd/bee.lua/bee/nonstd/fmt/core.h:1426:41: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'Context'
FMT_CONSTEXPR basic_format_arg<Context> make_arg(const T& value) {
                                        ^
4 errors generated.
[18/58] Compile C++ build/macos/obj/source_bee/format.objm.obj
ninja: build stopped: subcommand failed.
error: builder for '/nix/store/ssxb50ahrdw2034j4vbbf1wpmka9mg5p-sumneko-lua-language-server-2.5.6.drv' failed with exit code 1;
       last 10 log lines:
       >                            ^
       > 3rd/bee.lua/bee/fsevent/fsevent_osx.cpp:163:26: note: in instantiation of function template specialization 'fmt::v7::format<char [12], int &, char, 0>' requested here
       >                     std::format("remove `{}`", arg.m_id)
       >                          ^
       > 3rd/bee.lua/bee/nonstd/fmt/core.h:1426:41: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'Context'
       > FMT_CONSTEXPR basic_format_arg<Context> make_arg(const T& value) {
       >                                         ^
       > 4 errors generated.
       > [18/58] Compile C++ build/macos/obj/source_bee/format.objm.obj
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/ssxb50ahrdw2034j4vbbf1wpmka9mg5p-sumneko-lua-language-server-2.5.6.drv'.
```

</div>
</details>

### Steps To Reproduce

1. Run below in nixpkgs:
    ```
    $ nix-build -A sumneko-lua-language-server
    ```

### Expected behavior

Builds sumneko-lua-language-server successfully.

### Notify maintainers

@mjlbach 

### Metadata

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"nixpkgs-22.05pre343295.adf7f03d3bf"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/development/tools/sumneko-lua-language-server/default.nix
+++ b/pkgs/development/tools/sumneko-lua-language-server/default.nix
@@ -4,13 +4,13 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "sumneko-lua-language-server";
-  version = "2.5.6";
+  version = "2.6.0";
 
   src = fetchFromGitHub {
     owner = "sumneko";
     repo = "lua-language-server";
     rev = version;
-    sha256 = "sha256-dSj3wNbQghiGfqe7dNDbWnbXYLSiG+0mYv2yFmGsAc8=";
+    sha256 = "sha256-8Vfk6B85anlUf09cc08hOGujbcVCMqgEJ1PTxX6llrk=";
     fetchSubmodules = true;
   };
 

